Refactor `ReaderTopicService.currentTopic`

crazytonyli opened the pull request:
This PR replaces the `currentTopic` property in the interface file with a safer getter (which requires a context to perform the query) and a setter (whose implementation isn't changed).

During making this change, I noticed the `ReaderTopicDidChangeNotification` isn't observed by any code, so I deleted it for good—why bother sending this notification if no one observes it.

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
None.

3. What automated tests I added (or what prevented me from doing so)
None.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

crazytonyli committed:
Remove an notification which doesn't have any observer

crazytonyli committed:
Query current topic from given context

mokagio commented on WordPress/Classes/Services/ReaderTopicService.m:
```diff
@@ -126,10 +124,6 @@ - (void)setCurrentTopic:(ReaderAbstractTopic *)topic
-        
-        dispatch_async(dispatch_get_main_queue(), ^{
-            [[NSNotificationCenter defaultCenter] postNotificationName:ReaderTopicDidChangeNotification object:nil]; 
-        });
```
Nitpick. The `[NSUserDefaults resetStandardUserDefaults];` call is duplicated between the two `if` branches. It could be extracted after the `if`.

crazytonyli replied:
Good catch! I didn't notice that. The method calls has been removed in https://github.com/wordpress-mobile/WordPress-iOS/pull/20137/commits/5bcec77be8980b9901dc2dd69a6cca39094c45b1 since [the documentation](https://developer.apple.com/documentation/foundation/nsuserdefaults/1407708-resetstandarduserdefaults) says this method "shouldn't be used".

mokagio replied:
Ah! Even better!

mokagio reviewed (approved):
Thank you for removing the notification. I think it was a good YAGNI application. _If_ we'll ever need a mechanism to know when the reader topic changed, we'll implement one (hopefully of better, or at least equal, quality).

crazytonyli committed:
Remove `NSUserDefault.resetStandardUserDefaults` calls

The documentation says "this method has no effect and shouldn't be
used".

https://developer.apple.com/documentation/foundation/nsuserdefaults/1407708-resetstandarduserdefaults
